src/server/pkg/backoff/retry.go

```diff
@@ -49,3 +52,16 @@ func RetryNotify(operation Operation, b BackOff, notify Notify) error {
 		time.Sleep(next)
 	}
 }
+
+// RetryUntilCancel runs the given callback repeatedly until it returns an
```

Comment:
This comment seems a bit misleading. If the callback returns a nil error, then it returns immediately right? I guess based on the comments above and `Retry` in the name it might be easy to infer the actual functioning, but reading the comment without that context could be misleading. This function can also return if the Backoff stops as well (without the context being cancelled), so that should probably be mentioned. I think both off these issues can be addressed by making it clear that it functions like `RetryNotify` with the extra context cancelled check (then maybe change the `RetryNotify` comment to connect it back to the `Retry` function). Also, it seems a bit weird to have the context check in a notify callback. This seems like something that should be in the `RetryNotify` function, but I can understand how that would be painful to update at all the call sites. I think it would make more sense to refactor the loop in `RetryNotify` into its own non-exported function then have the `RetryNotify` and `RetryUntilCancel` functions call that. So, the ctx check would happen for each iteration.